source/distributed/tria.cc

```diff
@@ -3122,6 +3122,204 @@ namespace parallel
 
 
 
+    namespace
+    {
+      template <typename ITERATOR>
+      void
+      identify_periodic_vertices_recursively(const GridTools::PeriodicFacePair<ITERATOR> &periodic,
+                                             std::vector<unsigned int> &topological_vertex_numbering)
+      {
+        // for hanging nodes we will consider all necessary coupling already
```

Comment:
there are three cases:
1. both have children: you recurse, but you also drop out of the if and identify vertices (is that necessary?)
2. both have no children: you identify the vertices (okay)
3. one of them has children: here you end up identifying the wrong vertices, I think.